paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1174 @@
+import com.android.resources.ResourceType.DRAWABLE
+import com.android.resources.ResourceType.ID
+import com.android.resources.ResourceType.INTERPOLATOR
+import com.android.resources.ResourceType.LAYOUT
+import com.android.resources.ResourceType.MENU
+import com.android.resources.ResourceType.MIPMAP
+import com.android.resources.ResourceType.PLURALS
+import com.android.resources.ResourceType.PUBLIC
+import com.android.resources.ResourceType.STRING
+import com.android.resources.ResourceType.STYLE
+import com.android.resources.ResourceType.STYLEABLE
+import com.android.resources.ResourceType.TRANSITION
+import com.android.resources.ResourceVisibility
+import com.android.resources.ResourceVisibility.PRIVATE
+import com.android.utils.SdkUtils
+import com.android.utils.XmlUtils
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.Table
+import com.google.common.collect.Tables
+import org.kxml2.io.KXmlParser
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import java.io.BufferedInputStream
+import java.io.File
+import java.io.IOException
+import java.nio.file.FileVisitResult
+import java.nio.file.FileVisitor
+import java.nio.file.Path
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.ArrayDeque
+import java.util.Deque
+import java.util.EnumMap
+import java.util.EnumSet
+import java.util.zip.ZipFile
+
+/**
+ * Ported from: [RepositoryLoader.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/RepositoryLoader.java)
+ */
+abstract class RepositoryLoader<T : LoadableResourceRepository>(
+  val resourceDirectoryOrFile: Path,
+  val resourceFilesAndFolders: Collection<PathString>?,
+  val namespace: ResourceNamespace
+) : FileFilter {
+  private val logger = PaparazziLogger()
+  private val fileFilter =
+    PatternBasedFileFilter(AndroidAaptIgnore(System.getenv(ANDROID_AAPT_IGNORE)))
+  private val parser = ValueResourceXmlParser()
+  private val urlParser = ResourceUrlParser()
+  private val textExtractor: XmlTextExtractor = XmlTextExtractor()
+  private val resourceDirectoryOrFilePath: PathString = PathString(resourceDirectoryOrFile)
+  private val loadingFromZipArchive = isZipArchive(resourceDirectoryOrFile)
+  protected var zipFile: ZipFile? = null
+
+  /** Cache of FolderConfiguration instances, keyed by qualifier strings (see [FolderConfiguration.getQualifierString]).  */
+  protected val folderConfigCache: MutableMap<String, FolderConfiguration> = hashMapOf()
+  private val configCache: MutableMap<FolderConfiguration, RepositoryConfiguration> = hashMapOf()
+
+  private val valueFileResources: Table<ResourceType, String, BasicResourceItem> =
+    Tables.newCustomTable(
+      EnumMap(
+        ResourceType::class.java
+      )
+    ) { LinkedHashMap() }
+
+  fun loadRepositoryContents(repository: T) {
+    if (loadingFromZipArchive) {
```

Comment:
@jrodbx Could you review the loadResourceFile parsing flow? Details are just copied from original, just need to review overall flow to see anything confuse you? Thanks